cupy/polynomial/polynomial.py

```diff
@@ -0,0 +1,27 @@
+import cupy
+
+
+def polyvander(x, deg):
+    """Computes the Vandermonde matrix of given degree.
+
+    Args:
+        x (cupy.ndarray): array of points
+        deg (int): degree of the resulting matrix.
+
+    Returns:
+        cupy.ndarray: The Vandermonde matrix
+
+    .. seealso:: :func:`numpy.polynomial.polynomial.polyvander`
+
+    """
+    deg.__index__()
```

Comment:
What is this?

Reply:
for float with integral value case

Reply:
For `float` type `deg` case?

Reply:
yes

Reply:
Hmm, `float` object has no attribute `__index__`. What do you mean?

Reply:
I meant **`operator.index(a)`** which is equivalent to `a.__index__()` acoording to [this](https://docs.python.org/3/library/operator.html#operator.index)

Reply:
Will I need to implement _deprecate_as_int() as described [here](https://fossies.org/linux/numpy/numpy/polynomial/polyutils.py) L766

Reply:
I have already put it in `polyutils` to have the same behaviour as numpy. If it is not needed, we can support integers and raise `TypeError `otherwise while eliminating  `_deprecate_as_int()`